src/main/java/org/apache/cassandra/sidecar/routes/ListSnapshotFilesHandler.java

```diff
@@ -146,6 +160,12 @@ private ListSnapshotFilesResponse buildResponse(String host,
         return response;
     }
 
+    private boolean isSecondaryIndexFile(Path path)
+    {
+        return path.getNameCount() >= 4 &&
+               path.getName(path.getNameCount() - 4).endsWith(SNAPSHOTS_DIR_NAME);
```

Comment:
I thought about that, but it's unlikely that this will occur since the table appends the UUID to the directory name. In any case, I am thinking we can also validate for the `keyspace`, by changing the signature to this:
```java
    private boolean isSecondaryIndexFile(Path path, String keyspace)
    {
        return path.getNameCount() >= 6 &&
               path.getName(path.getNameCount() - 6).endsWith(keyspace) &&
               path.getName(path.getNameCount() - 4).endsWith(SNAPSHOTS_DIR_NAME);
    }
```
What do you think?

Reply:
discussed offline and we decided it's best to have `SnapshotPathBuilder#listSnapshotDirectory` return a List of `SnapshotFile` objects instead of returning a list of `Pair` objects.